sources/core/objects/Human.cpp

```diff
@@ -359,27 +375,36 @@ void Human::Live()
         --lay_timer_;
     }
 
-    if (health_ < 0)
+    if (CalculateHealth() < 0)
     {
         if (!lying_)
         {
             SetLying(true);
         }
-        if (health_ >= -100)
+        if (CalculateHealth() >= -10000)
         {
-            --health_;
+            AddSuffocationDamage(100);
             if (GetRand() % 4 == 0 && ((MAIN_TICK % 4) == 0))
             {
-                GetGame().GetChat().PostSimpleText(name + " gasps!", owner->GetId());
+                GetGame().GetChat().PostSimpleText(QString("%1 gasps!").arg(name), owner->GetId());
             }
         }
     }
-    if (health_ < -100 && !dead_)
+    if (CalculateHealth() < -10000 && !dead_)
     {
         OnDeath();
     }
 }
 
+void Human::Regeneration()
+{
+    int healed = suffocation_damage_ / 2 > 0 ? suffocation_damage_ / 2 : 1;
```

Comment:
I think, ternary operator should be avoided in non-trivial expressions - because the operator is hard to read.
Also, it is better to use getters even inside of class.
So overall it will be better:
```
if (GetSuffocationDamage() <= 0)
{
    return;
}
int healed = qMax(1, GetSuffocationDamage() / 2);
AddSuffocationDamage(-1 * healed);
```